pandas/core/dtypes/cast.py

```diff
@@ -793,10 +793,10 @@ def soft_convert_objects(values, datetime=True, numeric=True, timedelta=True,
         # Immediate return if coerce
         if datetime:
             from pandas import to_datetime
-            return to_datetime(values, errors='coerce', box=False)
+            return to_datetime(values, errors='coerce').to_numpy()
```

Comment:
we *may* not actually need to do this as the caller might be more friendly now-a-days

Reply:
You mean that we may not need to convert to ndarray?

Currently, one test fails due to a failed assertion in an internal function: pandas/core/internals/blocks.py:2120
The failing test is series.test_internals.TestSeriesInternals.test_convert

Reply:
of its just an internal function, then pls just change it.